service/storage_service.cc

```diff
@@ -6682,5 +6683,11 @@ future<> storage_service::node_ops_abort_thread() {
     __builtin_unreachable();
 }
 
+future<> storage_service::start_maintanance_mode() {
+    return mutate_token_metadata([] (mutable_token_metadata_ptr token_metadata) {
+        return token_metadata->update_normal_tokens({ dht::token{} }, utils::fb_utilities::get_broadcast_address());
+    });
```

Comment:
What about topology over raft that may just recreate the token_metadata again?

Reply:
When might this happen? In the maintenance mode group0 is not started and communication with other nodes is not working.

Reply:
If you do not start group0 it cannot, but you do not need communication from other nodes for group0 to re-create token metadata.

Reply:
But if you do not start group0 and the system thinks that if is in raft mode it is a disaster waiting to happen. I saw that you added protection to the guard, but now we have to be careful to protect any places that access group0 outside of it current and future.

Reply:
As I understood, setting `raft_group0_client::_upgrade_state` to `group0_upgrade_state::recovery` protects us from such a disaster.